tla-types/src/main/scala/at/forsyte/apalache/tla/typecomp/ScopedBuilder.scala

```diff
@@ -25,9 +27,114 @@ import at.forsyte.apalache.tla.typecomp.subbuilder.{
  *     not be scope-correct, since x would have to be typed as Int within the scope defined by this \A operator. Thus,
  *     such an expression cannot be constructed by the builder.
  *
+ * Note that the guarantees are void if `useTrustedEx` is used.
+ *
  * @author
  *   Jure Kukovec
  */
 class ScopedBuilder
-    extends BaseBuilder with BoolBuilder with ArithmeticBuilder with SetBuilder with SeqBuilder
-    with LiteralAndNameBuilder
+    extends BaseBuilder with BoolBuilder with ArithmeticBuilder with SetBuilder with SeqBuilder with ActionBuilder
+    with LiteralAndNameBuilder {
+
+  /*
+  To use the builder outside of testing scenarios, where the expressions aren't written from scratch,
+  it is necessary to be able to mark certain expressions as "trusted", e.g. when transforming and recombining invariants,
+  or parts of Init/Next.
+  While `build` is safe as a unidirectional implicit conversion from TBuilderInstruction to TLaEx,
+  the inverse, `useTrustedEx`, needs to be explicit, to stress the fact that it should be used rarely,
+  typically at most once per transformation, on the initial input.
+   */
+  /**
+   * Creates a `TBuilderInstruction` from a precomputed `TlaEx`.
```

Comment:
Can we just include this paragraph in the Scaladoc below?

Reply:
We could, but the purpose of splitting docs was to keep the Scaladoc lean and limited to descriptions, whereas the normal doc would offer an explanation to devs of why/how the decision was made, which someone using the builder needn't care about.

Reply:
Why do we need the Scaladoc to be lean and limited?
It's [quite common](http://hg.openjdk.java.net/jdk8/jdk8/jdk/file/f92ab6dbbff8/src/share/classes/java/lang/Long.java#l465) to have longer doc comments if necessary.

> To use the builder outside of testing scenarios [...]

is definitely relevant to someone calling the method.

Reply:
I also do not get the point of hiding the docs in a place where but a few people can see it. You could mark the paragraph as a detailed description.